src/main/java/com/mwt/explorers/EpsilonGreedyExplorer.java

```diff
@@ -45,22 +45,23 @@ public void updatePolicy(Policy<T> newPolicy) {
     this.defaultPolicy = newPolicy;
   }
 
-  public ChosenAction chooseAction(long saltedSeed, T context) {
+  public DecisionTuple chooseAction(long saltedSeed, T context) {
+    int numActionsForContext = getNumActions(context);
     Random random = new Random(saltedSeed);
```

Comment:
The C++ version is explicit.

https://github.com/JohnLangford/vowpal_wabbit/blob/master/explore/static/utility.h#L242

-John

On 06/19/2015 10:28 AM, Luong Hoang wrote:
>
> In src/main/java/com/mwt/explorers/EpsilonGreedyExplorer.java 
> <https://github.com/multiworldtesting/explore-java/pull/2#discussion_r32833752>:
>
> > @@ -45,22 +45,23 @@ public void updatePolicy(Policy<T> newPolicy) {
> >      this.defaultPolicy = newPolicy;
> >    }
> >
> > -  public ChosenAction chooseAction(long saltedSeed, T context) {
> > +  public DecisionTuple chooseAction(long saltedSeed, T context) {
> > +    int numActionsForContext = getNumActions(context);
> >      Random random = new Random(saltedSeed);
>
> I haven't got to testing yet but it definitely makes correctness 
> verification a lot easier. The C++ version is using a built-in random 
> generator I believe, so we'll need to converge on one, possibly 
> something already in VW since we're using the hash there.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/multiworldtesting/explore-java/pull/2/files#r32833752>.
>